source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs

```
@@ -102,12 +102,19 @@ public void Update(ChargePeriod newChargePeriod)
                 throw new InvalidOperationException("Charge update must not have bound end date.");
             }
 
+            var stopPeriod = _periods.FirstOrDefault(p => p.StartDateTime == p.EndDateTime
+                                                          && p.StartDateTime > newChargePeriod.StartDateTime);
```

if `p.StartDateTime == p.EndDateTime` and we are in the update-scenario, I think `p.StartDateTime > newChargePeriod.StartDateTime` is always true? - or that would be inconsistent data, that should result in throw `InvalidOperationException`.